src/sc/sidechaintypes.h

```diff
@@ -4,198 +4,252 @@
-    ~CompressedFieldElementConfig() = default;
+    FieldElementCertificateFieldConfig(int32_t nBitsIn);
+    FieldElementCertificateFieldConfig(const FieldElementCertificateFieldConfig& rhs) = default;
+    ~FieldElementCertificateFieldConfig() = default;
+
+    //For serialization only, which requires the default ctor. No checkValid call here
+    FieldElementCertificateFieldConfig(): CustomCertificateFieldConfig(), nBits(0) {}
 
-    int32_t getBitSize() const override; //TENTATIVE IMPLEMENTATION, BEFORE ACTUAL ONE
+    bool IsValid() const override final;
 
     ADD_SERIALIZE_METHODS;
     template <typename Stream, typename Operation>
     inline void SerializationOp(Stream& s, Operation ser_action, int nType, int nVersion) {
         READWRITE(nBits);
-
-        if (!isBitsLenghtValid())
-            throw std::invalid_argument("CompressedFieldElementConfig size must be strictly positive");
     }
 
-    bool operator==(const CompressedFieldElementConfig& rhs) const {
+    int32_t getBitSize() const;
+
+    bool operator==(const FieldElementCertificateFieldConfig& rhs) const {
         return (this->nBits == rhs.nBits);
     }
 
-    bool operator!=(const CompressedFieldElementConfig& rhs) const {
+    bool operator!=(const FieldElementCertificateFieldConfig& rhs) const {
         return !(*this == rhs);
     }
 
-    friend std::ostream& operator<<(std::ostream& os, const CompressedFieldElementConfig& r) {
+    friend std::ostream& operator<<(std::ostream& os, const FieldElementCertificateFieldConfig& r) {
         os << r.nBits;
         return os;
     }
 };
 
-class CompressedMerkleTreeConfig : public CustomFieldConfig
+class BitVectorCertificateFieldConfig : public CustomCertificateFieldConfig
 {
 private:
-    int32_t treeHeight;
-    static const int32_t MAX_TREE_HEIGHT = log2(std::numeric_limits<int32_t>::max()); //TENTATIVE IMPLEMENTATION, BEFORE ACTUAL ONE
-    bool isTreeHeightValid(); //TENTATIVE IMPLEMENTATION, BEFORE ACTUAL ONE
+    int32_t bitVectorSizeBits;
+    int32_t maxCompressedSizeBytes;
 
 public:
-    CompressedMerkleTreeConfig(int32_t treeHeightIn);
-    CompressedMerkleTreeConfig(): CustomFieldConfig(), treeHeight(-1) {} //for serialization only, which requires the default ctor
-    ~CompressedMerkleTreeConfig() = default;
+    BitVectorCertificateFieldConfig(int32_t bitVectorSizeBits, int32_t maxCompressedSizeBytes);
+    ~BitVectorCertificateFieldConfig() = default;
 
-    int32_t getBitSize() const override; //TENTATIVE IMPLEMENTATION, BEFORE ACTUAL ONE
+    //for serialization only, which requires the default ctor. No checkValid call here
+    BitVectorCertificateFieldConfig(): CustomCertificateFieldConfig(), bitVectorSizeBits(-1), maxCompressedSizeBytes(-1) {}
+
+    static const int32_t MAX_BIT_VECTOR_SIZE_BITS = 1000192;
+    static const int32_t MAX_COMPRESSED_SIZE_BYTES = MAX_BIT_VECTOR_SIZE_BITS / 8;
```

Comment:
If some SC will specify a config with max allowed size of `MAX_COMPRESSED_SIZE_BYTES`, then a particular certificate may contain a bitvector with the size of > 15mb, which exceed all possible limits (tx size, block size).

Reply:
Redone calculation together, math sticks. However, as you noticed, current value for MAX_COMPRESSED_SIZE_BYTES is larger than MAX_TX_SIZE (100000 bytes) consensus parameter. @albertog78 we should prolly retune the MAX_COMPRESSED_SIZE_BYTES

Reply:
As discussed with Algaro, we are going to introduce a new consensus parameter MAX_CERT_SIZE, set to 150000

Reply:
fine